File: ui/src/app/blog/article.ts
import {SafeHtml} from "@angular/platform-browser";

export class Article {
  id: number;
  path: string;
  date: string;
  title: string;
  introduction: string;
  metaDescription: string;
  metaKeywords: string;
  body: string;
  safeHTML: SafeHtml;
}

export class ArticleListItem {
  id: number;
  path: string;
  title: string;
  date: string;
  body: string;
}

export class ArticleList {
  items: ArticleListItem[];
  page: number;
  page_count: number;
  per_page: number;
  total_count: number;
}

File: models/article.go
package models

import (
	"github.com/go-ozzo/ozzo-validation"
)

type Article struct {
	Id              int    `json:"id"`
	CategoryId      int    `json:"categoryId"`
	Path            string `json:"path"`
	Date            string `json:"date"`
	Title           string `json:"title"`
	MetaDescription string `json:"metaDescription"`
	MetaKeywords    string `json:"metaKeywords"`
	Introduction    string `json:"introduction"`
	Body            string `json:"body"`
}

func (m Article) Validate() error {
	return validation.ValidateStruct(&m,
		//validation.Field(&m.Title, validation.Required, validation.Length(1, 150)),
		//validation.Field(&m.Body, validation.Required, validation.Length(1, 1550)),
	)
}
